### PR TITLE
feat: support cache sealer image

### LINKED ISSUE #2010: Optimize the efficiency of sealer file distribution.
## Issue Description
 For nodes that have passed the mirror layer of the corresponding version, the transmission should be skipped.

Type: *feature request*

### Describe what feature you want

I think you can implement this feat by this way:
1. when sealer run, it record current cluster image's sha256 id on every target node
2. when sealer run again, it check if target node record current cluster image's sha256, if yes, skip these nodes,
3. when sealer delete, do not clean files by default. And provide a flag like "--prune", if true, will clean record first and then remove all files from these nodes.

### PR DESCRIPTION
**Pull request by kakaZhou719**
### Describe what this PR does / why we need it

add flag  `ignore-cache` for run, apply, join, scale-up, rollback, indicates that whether use cache when distribute sealer image, default is false.

### Does this pull request fix one issue?

Fixes #2010

### CODE REVIEW

**Comment by Stevent-fei** [comment]
pls rebase commits

File: cmd/sealer/cmd/cluster/delete.go [hunk 1]
```diff
@@ -89,7 +89,7 @@ func NewDeleteCmd() *cobra.Command {
 	deleteCmd.Flags().StringSliceVarP(&deleteFlags.CustomEnv, "env", "e", []string{}, "set custom environment variables")
 	deleteCmd.Flags().BoolVar(&deleteFlags.ForceDelete, "force", false, "We also can input an --force flag to delete cluster by force")
 	deleteCmd.Flags().BoolVarP(&deleteFlags.DeleteAll, "all", "a", false, "this flags is for delete the entire cluster, default is false")
-	deleteCmd.Flags().BoolVarP(&imagedistributor.IsPrune, "prune", "p", true, "this flags is for delete all cluster rootfs, default is true")
+	deleteCmd.Flags().BoolVarP(&deleteFlags.Prune, "prune", "p", false, "this flags is for delete all cluster rootfs, default is true")
```
Comment (Stevent-fei) [inline cmd/sealer/cmd/cluster/delete.go:92]: default is true?

File: cmd/sealer/cmd/cluster/apply.go [hunk 2]
```diff
@@ -164,6 +166,7 @@ func NewApplyCmd() *cobra.Command {
 	applyCmd.Flags().Uint16Var(&applyFlags.Port, "port", 22, "set the sshd service port number for the server (default port: 22)")
 	applyCmd.Flags().StringVar(&applyFlags.Pk, "pk", filepath.Join(common.GetHomeDir(), ".ssh", "id_rsa"), "set baremetal server private key")
 	applyCmd.Flags().StringVar(&applyFlags.PkPassword, "pk-passwd", "", "set baremetal server private key password")
+	applyCmd.Flags().BoolVar(&applyFlags.UseCache, "use-cache", true, "whether use cache when distribute sealer image, default is true.")
```
Comment (Starnop) [inline cmd/sealer/cmd/cluster/apply.go:169]: --no-cache maybe is better

File: cmd/sealer/cmd/cluster/join.go [hunk 3]
```diff
@@ -110,5 +110,7 @@ func NewJoinCmd() *cobra.Command {
 	joinCmd.Flags().StringSliceVarP(&joinFlags.CustomEnv, "env", "e", []string{}, "set custom environment variables")
 	joinCmd.Flags().StringVarP(&joinFlags.Masters, "masters", "m", "", "set Count or IPList to masters")
 	joinCmd.Flags().StringVarP(&joinFlags.Nodes, "nodes", "n", "", "set Count or IPList to nodes")
+	joinCmd.Flags().BoolVar(&joinFlags.UseCache, "use-cache", true, "whether use cache when distribute sealer image, default is true.")
```
Comment (Starnop) [inline cmd/sealer/cmd/cluster/join.go:113]: ditto

File: cmd/sealer/cmd/cluster/rollback.go [hunk 4]
```diff
@@ -103,6 +103,7 @@ func NewRollbackCmd() *cobra.Command {
 
 	rollbackFlags = &types.RollbackFlags{}
 	rollbackCmd.Flags().StringSliceVar(&rollbackFlags.AppNames, "apps", nil, "override default AppNames of sealer image")
+	rollbackCmd.Flags().BoolVar(&rollbackFlags.UseCache, "use-cache", true, "whether use cache when distribute sealer image, default is true.")
```
Comment (Starnop) [inline cmd/sealer/cmd/cluster/rollback.go:106]: ditto

**Review by Starnop — APPROVED** [review]
LGTM

**Pull request merged**